internal/eventstore/usecases/retrieve_events.go

```diff
@@ -89,3 +111,14 @@ func (u *RetrieveEventsUseCase) Run(ctx context.Context) error {
 
 	return nil
 }
+
+func (u *RetrieveEventsUseCase) load(ctx context.Context, storeQueries []*es.StoreQuery) (es.EventStreamReceiver, error) {
+	switch u.gate {
```

Comment:
Why this gate? Couldn't you just make this decision based on if there is more than one filter?

Reply:
Yeah i wasn't sure about this one either but I thought maybe just maybe it made sense in the future to support more gates like and, xor, etc...., What do you think?

But yeah you are right I can for sure base the decision on the number filters.

Reply:
If we need something like this in the future we can still refactor it like that. It always makes sense to keep things simple and don't plan for use-cases that might never come. ;-)

Reply:
Alrighty 👍